instrumentation/opentelemetry_finch/lib/opentelemetry_finch.ex

```diff
@@ -44,8 +44,17 @@ defmodule OpentelemetryFinch do
 
     status =
       case meta.result do
-        {:ok, response} -> response.status
-        _ -> 0
+        {:ok, response} when is_map(response) ->
```

Comment:
The main issue was here. Not every response contains a status key as it depends on the implementation of the streaming callback functions.

A fallback case is also added. In this fix, I'm not trying to come up with a general solution but just one that works for the patterns I saw. The tests use callbacks that return values in the patterns I was already working with.

Reply:
I guess you could just match on the presence of the status:

```elixir
case meta.result do
  {:ok, %{status: status}} when integer(status) -> status
  {:ok, {_request, %{status: status}}} when integer(status) -> status
  {:ok, {status, _headers, _body}} when is_integer(status) -> status
  _ -> 0
end
```

Because there is a risk that even when `response` is a map, `response.status` may not be set. As far as I can tell that part of the event is the stream accumulator and it could be anything, any term.

Reply:
(edit fixed my clauses :') )